subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_security.c

```diff
@@ -311,24 +359,28 @@ static int server_created(uint16_t id)
 static int security_deleted(uint16_t id)
 {
 	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_SERVER_URI_ID);
+	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_BOOTSTRAP_FLAG_ID);
 	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_MODE_ID);
 	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_CLIENT_PK_ID);
 	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_SERVER_PK_ID);
-	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_SECRET_KEY_ID);
 	delete_from_storage(LWM2M_OBJECT_SECURITY_ID, id, SECURITY_SHORT_SERVER_ID);
 
+	/* This happens when new bootstrap starts, so remove any
+	 * flags that I have new keys because those are now outdated
```

Comment:
OK, sure.

So the point is that flags like `bootstrap_settings_loaded_inst` and `have_permanently_stored_keys` might trigger key update. But data for those are now removed, because someone (most probably the bootstrap server) removed the security instances.

This was a last bug on my code that after a bootstrap that did not update BS key, I still had the flag that indicated that the key is updated, so it ended up writing garbage to  the modem.

I'll try to reword the comment a bit.

Reply:
Actually, now I modified the code a bit. On line 174 I'm already resetting the boostrap flag, so this is not needed anymore.

I dropped the comment as now is obvious what is done here.